### PR TITLE
Indexer: GetEnvelopeHeight: return 0 for process without results

### PR DESCRIPTION
**Pull request by NateWilliams2**

### CODE REVIEW

File: vochain/scrutinizer/vote.go [hunk 1]
```diff
@@ -224,6 +224,11 @@ func (s *Scrutinizer) GetEnvelopeHeight(processID []byte) (uint64, error) {
 	// If not cached, make the expensive query
 	results := &indexertypes.Results{}
 	if err := s.db.FindOne(results, badgerhold.Where(badgerhold.Key).Eq(processID)); err != nil {
+		// If results are not found, return envelope height of 0.
```
Comment (p4u) [inline vochain/scrutinizer/vote.go]: With this approach we cannot differentiate between canceled processes and not existing processes.

I think return en ErrNotFoundInDatabase if a process is Canceled is the right approach. There are not results and that's what we are returning.

We could modify the scrutinizer it norder to not remove Canceled processes but set it to 0 instead. However I still don't understand what's the problem behind returnin ErrNotFoundInDatabase. Can you clarify?

Comment (NateWilliams2) [reply vochain/scrutinizer/vote.go]: To me, in the case of a cancelled process, it seems really messy to return an ErrNotFoundInDatabase when querying for envelope count, if the envelope count is in fact known. Whoever is invoking the GetEnvelopeCount shouldn't need to expect an error as standard, non-error behavior.

Comment (p4u) [reply vochain/scrutinizer/vote.go]: Ok, then let's modify the scrutinizer logic to not remove the results entry (just clean Votes but let EnvelopeHeight and Weight) and set Final to true. Does that work?

Comment (NateWilliams2) [reply vochain/scrutinizer/vote.go]: that works for me! I'll implement that

File: vochain/scrutinizer/process.go [hunk 2]
```diff
@@ -385,7 +385,21 @@ func (s *Scrutinizer) updateProcess(pid []byte) error {
...
+					results, ok := record.(*indexertypes.Results)
+					if !ok {
+						return fmt.Errorf("record isn't the correct type! Wanted Result, got %T", record)
+					}
+					// On cancelled process, remove all results except for envelope height, weight, pid
+					results.Votes = [][]*big.Int{}
+					results.EnvelopeType = &models.EnvelopeType{}
+					results.VoteOpts = &models.ProcessVoteOptions{}
+					results.Signatures = []types.HexBytes{}
+					results.Final = *types.False
```
Comment (p4u) [inline vochain/scrutinizer/process.go]: That's weird, I'd just set it to "false"

**Review by p4u — APPROVED** [review]

**Pull request merged**